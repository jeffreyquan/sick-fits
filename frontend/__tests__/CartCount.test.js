import { render, screen } from "@testing-library/react";
import wait from "waait";
import CardCount from "../components/CartCount";

describe("<CardCount/>", () => {
  it("renders", () => {
    render(<CardCount count={10} />);
  });

  it("matches snapshot", () => {
    const { container } = render(<CardCount count={11} />);
    expect(container).toMatchSnapshot();
  });

  it("updates via props", async () => {
    const { container, rerender } = render(<CardCount count={11} />);
    expect(container).toHaveTextContent("11");
    // Update the props
    rerender(<CardCount count="12" />);
    // await for __ ms
    // await wait(400);
    expect(container).toHaveTextContent("1211");
    await screen.findByText("12");
    expect(container).toHaveTextContent("12");
    expect(container).toMatchSnapshot();
  });
});
