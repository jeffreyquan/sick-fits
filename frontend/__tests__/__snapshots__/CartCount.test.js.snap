// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardCount/> matches snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 gwXiGr"
  >
    <div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 dgpiUM count"
      >
        11
      </div>
    </div>
  </span>
</div>
`;

exports[`<CardCount/> updates via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 gwXiGr"
  >
    <div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 dgpiUM count count-enter count-enter-active"
      >
        12
      </div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 dgpiUM count count-exit count-exit-active"
      >
        11
      </div>
    </div>
  </span>
</div>
`;
